x-pack/plugins/security_solution_serverless/server/cloud_security/cloud_security_metering_task.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  CNVM_POLICY_TEMPLATE,
+  CSPM_POLICY_TEMPLATE,
+  KSPM_POLICY_TEMPLATE,
+  LATEST_FINDINGS_INDEX_PATTERN,
+  LATEST_FINDINGS_RETENTION_POLICY,
+  LATEST_VULNERABILITIES_INDEX_PATTERN,
+  LATEST_VULNERABILITIES_RETENTION_POLICY,
+} from '@kbn/cloud-security-posture-plugin/common/constants';
+import type { UsageRecord } from '../types';
+
+import {
+  AGGREGATION_PRECISION_THRESHOLD,
+  CLOUD_SECURITY_TASK_TYPE,
+} from './cloud_security_metering';
+import { cloudSecurityMetringTaskProperties } from './cloud_security_metering_task_config';
+import type {
+  CloudSecurityMeteringCallbackInput,
+  PostureType,
+  ResourceCountAggregation,
+} from './types';
+
+const queryParams = {
+  [CSPM_POLICY_TEMPLATE]: {
+    index: LATEST_FINDINGS_INDEX_PATTERN,
+    timeRange: LATEST_FINDINGS_RETENTION_POLICY,
+  },
+  [KSPM_POLICY_TEMPLATE]: {
+    index: LATEST_FINDINGS_INDEX_PATTERN,
+    timeRange: LATEST_FINDINGS_RETENTION_POLICY,
```

Comment:
can be optimized to 5 hours

Reply:
it already discussed [here](https://github.com/elastic/kibana/pull/162469#discussion_r1279097915).